doc/admin-guide/plugins/filter_body.en.rst

```diff
@@ -0,0 +1,330 @@
+- The HTTP status is set to 403 Forbidden
+
+.. note::
+
+    For request body transforms, blocking occurs after body inspection begins.
+    The connection is closed rather than returning a clean HTTP 403 response
+    to the client.
+
+Add Header Action
+-----------------
+
+When the ``add_header`` action is configured, a custom header is added:
+
+- For request rules: The header is added to the request sent to the origin
+- For response rules: The header is added to the response sent to the client
+
+The header name and value are specified by ``add_header_name`` and
+``add_header_value`` configuration options.
+
+Example Configurations
+======================
+
+XXE Attack Detection
+--------------------
+
+Block XML requests containing XXE patterns::
+
+    rules:
+      - name: xxe_detection
+        direction: request
+        methods:
+          - POST
+          - PUT
+        headers:
+          - name: Content-Type
+            patterns:
+              - "application/xml"
+              - "text/xml"
+              - "application/xhtml+xml"
+        body_patterns:
+          - "<!ENTITY"
+          - "<!DOCTYPE"
+          - "SYSTEM"
+          - "PUBLIC"
+        action:
+          - log
+          - block
+
+SQL Injection Detection (Log Only)
+----------------------------------
+
+Log potential SQL injection attempts without blocking::
+
+    rules:
+      - name: sql_injection_detection
+        direction: request
+        methods:
+          - POST
+          - GET
+        max_content_length: 65536
+        body_patterns:
+          - "' OR '"
+          - "'; DROP"
+          - "UNION SELECT"
+          - "1=1"
+        action:
+          - log
+        add_header_name: X-Security-Warning
+        add_header_value: sql-injection-detected
+
+Sensitive Data Detection in Responses
+-------------------------------------
+
+Add a header when response contains sensitive patterns::
+
+    rules:
+      - name: sensitive_data_leak
+        direction: response
+        headers:
+          - name: Content-Type
+            patterns:
+              - "application/json"
+              - "text/html"
+        body_patterns:
+          - "password"
+          - "ssn"
+          - "credit_card"
+        action:
+          - log
+          - add_header
+        add_header_name: X-Data-Classification
+        add_header_value: sensitive
+
+Debugging
+=========
+
+To enable debug output for the plugin, configure debug tags in records.yaml::
+
+    records:
+      proxy.config.diags.debug.enabled: 1
+      proxy.config.diags.debug.tags: filter_body
+
+Debug output includes:
+
+- Configuration loading and rule parsing
+- Header matching results
+- Pattern match notifications
+- Action execution
+
+Limitations
+===========
+
+1. **Request blocking**: When blocking request bodies, the connection is closed
+   rather than returning a proper HTTP 403 response. This is a limitation of
+   request body transforms in |TS|.
```

Comment:
Also indicate here, as well as in the above note, that we cannot guarantee that an offensive body is blocked before it is sent to the origin.